src/ui/macstr_ui.py

```diff
@@ -0,0 +1,201 @@
+import sys
+import yaml
+from PyQt5 import QtCore, QtGui, QtWidgets
+
+
+class Ui_MainWindow(object):
+    def setupUi(self, MainWindow):
+
+        # loading in the JSON data
+        with open("../../notebooks/test_image_coords.json", "r") as contents:
+            image_data = yaml.safe_load(contents)
+            print(image_data["cell_image_1.tiff"])
+
+        # creating main GUI window
+        MainWindow.setObjectName("MainWindow")
+        MainWindow.resize(800, 600)
+
+        self.centralwidget = QtWidgets.QWidget(MainWindow)
+        self.centralwidget.setObjectName("centralwidget")
+
+        # ------------------------------
+        # setting up image canvas
+        # ------------------------------
+        self.image1 = QtWidgets.QLabel(self.centralwidget)
+        self.image1.setGeometry(QtCore.QRect(30, 10, 221, 191))
+        self.image1.setText("")
+        self.image1.setPixmap(
+            QtGui.QPixmap("7-stunning-cell-images-from-2017-294838-640x360.jpg")
```

Comment:
These are stock images that I have just downloaded to act as placeholders in order to present the GUI in this PR. These will be replaced once I implement GUI "events/actions"

Reply:
Thanks - even if they will be removed eventually, I recommend providing attribution while the images remain within the source code (they will persist in the git history and will appear in `main` for a time while work is underway). This also may serve to avoid copyright challenges in cases where the licensing is a possible area of dispute.

Reply:
Ahhhh good point. I'll update this